specification/indices/explain_data_lifecycle/IndicesExplainDataLifecycleRequest.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { RequestBase } from '@_types/Base'
+import { IndexName } from '@_types/common'
+import { Duration } from '@_types/Time'
+
+/**
+ * Retrieves information about the index's current DLM lifecycle, such as any potential encountered error, time since creation etc.
+ * @rest_spec_name indices.explain_data_lifecycle
+ * @since 8.8.0
+ * @stability experimental
+ */
+export interface Request extends RequestBase {
+  path_parts: {
+    index: IndexName
```

Comment:
It can accept both `.ds-my-data-stream-2023.05.05-000002,.ds-my-data-stream-2023.05.05-000001` and `.ds-my-data-stream-*`. So, I will change it.

Reply:
So, on a second look, indeed it can have one or more index names, but in other endpoints that exhibit the same behaviour we are using singular. Shouldn't we keep it inline with the rest of the APIs?

Reply:
Anywhere that allows one or more should use `Indices` rather than `IndexName` so the client can generate the "more than 1" case correctly. If you could update the other places you've found this that would be great :)

Reply:
Oh I was so wrong.... The other cases were correct, we were dealing with data streams so we were using `DataStreamNames`, I didn't notice the `s` before :). Fixed.